### PR TITLE
Use hearing_location.name for regional_office in hearings API

### LINKED ISSUE #11834: Hearings API regional_office should reflect hearing location
The Hearings API returns a record like:

```json
{
  "address":"100 SW Main Street Floor 2",
  "city":"Portland",
  "appeal":"701dc7a2-732e-4eca-9dc5-f024adf0719d",
  "facility_id":"vba_348",
  "participant_id":"31634112",
  "regional_office":"Seattle regional office",
  "room":"7",
  "scheduled_for":"2019-08-23T12:30:00.000-07:00",
  "state":"OR",
  "timezone":"America/Los_Angeles",
  "zip_code":"97204"
}
```

Is this correct?

### PR DESCRIPTION
**Pull request by pkarman**
Resolves #11834 

### Description
use the hearing_location.name as the `regional_office` value.

### CODE REVIEW

File: app/serializers/api/v2/hearing_serializer.rb [hunk 1]
```diff
@@ -18,7 +18,9 @@ class Api::V2::HearingSerializer
   attribute :participant_id do |hearing|
     hearing.appeal.veteran.participant_id
   end
-  attribute :regional_office, &:regional_office_name
+  attribute :regional_office do |hearing|
```
Comment (lomaxap) [inline app/serializers/api/v2/hearing_serializer.rb]: i think "regional office" has a specific meaning in the VA. e.g the hearing location can be a health center or an employment center. is there any concern there?

Comment (pkarman) [reply app/serializers/api/v2/hearing_serializer.rb]: that's a good point. we should rename it. I'll need to give the VEText folks a heads up.

File: app/serializers/api/v2/hearing_serializer.rb [hunk 2]
```diff
@@ -18,7 +18,9 @@ class Api::V2::HearingSerializer
...
+    hearing.hearing_location&.name
```
Comment (lomaxap) [inline app/serializers/api/v2/hearing_serializer.rb]: there are a few records that were created before hearing location mandatory/existed and the location is blank (#11802) so could do `hearing.hearing_location&.name || hearing.regional_office_name`

File: app/serializers/api/v2/hearing_serializer.rb (hunk 1 above)
Comment (pkarman) [reply app/serializers/api/v2/hearing_serializer.rb]: I added the new key name and marked the old as deprecated. We can remove it in a week. I emailed the VEText folks to that effect.

**Comment by pkarman** [comment]
thanks @lomaxap -- PR updated with your suggestions.

**Pull request merged**